src/api/js/src/high-level/utils.ts

```diff
@@ -0,0 +1,74 @@
+import { Z3AssertionError } from './types';
+
+function getAllProperties(obj: Record<string, any>) {
+  const properties = new Set<[any, string | symbol]>();
+  do {
+    for (const key of Reflect.ownKeys(obj)) {
+      properties.add([obj, key]);
+    }
+  } while ((obj = Reflect.getPrototypeOf(obj)!) && obj !== Object.prototype);
+  return properties;
+}
+
+// https://github.com/sindresorhus/auto-bind, MIT license
+// We copy it because it uses ESM instead of CommonJS
```

Comment:
`autoBind` is MIT licensed and so cannot be copy-pasted without including the license. But this shouldn't be necessary anyway, as mentioned above.